### PR TITLE
docs: name services: agoricNames, namesByAddress, board (WIP)

### LINKED ISSUE #807: document NameHub API, agoricNames, namesByAddress
As noted in https://github.com/Agoric/agoric-sdk/issues/8114 , 

Neither `namesByAddress` nor `myAddressNameAdmin` nor even the [NameHub](https://github.com/Agoric/agoric-sdk/blob/master/packages/vats/src/nameHub.js) API seems to be documented.

For example, `namesByAddress` and `myAddressNameAdmin`, as well as `agoricNames`, are missing from https://docs.agoric.com/reference/repl/

### PR DESCRIPTION
**Pull request by dckc**
fixes #807 

DRAFT until
 - [ ] add TOC entry (and decide whether `integrations/` is a reasonable place to put this)

stretch goals:
 - [ ] make code examples into tested snippets
    - [ ] by adding an example dapp that uses these namespaces
 - [ ] explain relationship between agoricNames and consume/produce patterns in permissioned contract deployment
 - [ ] reconcile with existing board docs under REPL
 - [ ] integrate with generated API doc? https://github.com/Agoric/agoric-sdk/pull/8720

### CODE REVIEW

File: main/guides/integration/name-services.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+ - `instance` - reserve, one for each psm, one for each priceAuthority, ... (see [deployed contracts](../zoe/actual-contracts/))
+
+Adding or updating an entry requires access to the corresponding _NameAdmin_ facet: `agoricNamesAdmin` or
+one of its descendants - most likely as part of [permissioned contract deployment](../coreeval/). For example:
+
+```js
+const { instance } = await E(zoe).startInstance(myNewContract);
+const { brands: { Thing: thingBrand } } = await E(zoe).getTerms(instance);
+const brandAdmin = E(agoricNamesAdmin).lookupAdmin('brand');
+await E(brandAdmin).update('Thing', thingBrand);
```
Comment (dckc) [inline main/guides/integration/name-services.md:42]: This is a _bad_ example. BLD staker governance should not approve meaningless names such as **Thing**.

Comment (dckc) [reply main/guides/integration/name-services.md:42]: Also TODO: note the risk of registering under a name such as **BTC** a brand for an asset that I just made up and that the market will _not_ value at 1 BTC.

File: main/guides/integration/name-services.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+```
+
+`agoricNames` is a _NameHub_. In general, `E(hub).lookup(key1, key2, ...rest)` is equivalent
+to `E(E(hub).lookup(key1)).lookup(key2, ...rest)`. The NameHub interface also includes `Map`-like
+methods `has`, `entries`, `keys`, and `values`.
+
+The keys of `agoricNames` include:
+ - `brand` - IST, BLD, Invitation (Zoe invitation brand), timer (a TimerBrand, not an ERTP brand)
+ - `issuer` - IST, BLD, Invitation, ...
+ - `vbankAsset` - IST, BLD, USDC, ATOM, ... - including info about corresponding cosmos denom
```
Comment (dckc) [inline main/guides/integration/name-services.md:30]: TODO: give the type of the values of the `vbankAsset` namehub

**Review by kbennett2000 — APPROVED** [review]

File: main/guides/integration/name-services.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+Also, a _deposit facet_ is installed at `E(namesByAddress).lookup(a, 'depositFacet')` so
+that you can send a payment, such as an invitation, to such a smart wallet using
+the `receive` method of the deposit facet:
+
+```js
+const bobAddress = 'agoric1...';
+const anInvitationPayment = await zcf.makeInvitation(...);
+const bobDepositFacet = E(namesByAddress).lookup(bobAddress, 'depositFacet');
+const amtReceived = await E(bobDepositFacet).receive(anInvitationPayment);
+```
```
Comment (dckc) [inline main/guides/integration/name-services.md:69]: ideally, these would be tested snippets of an example dapp

But dapp-offer-up doesn't use namesByAddress.

File: main/guides/integration/name-services.md (hunk 2 above)
Comment (dckc) [reply main/guides/integration/name-services.md:30]: and/or link to https://docs.agoric.com/guides/getting-started/contract-rpc.html#vbank-assets-and-cosmos-bank-balances

File: main/guides/integration/name-services.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
+# Name Services
+
+The Agoric chain has a few name services:
+ - `agoricNames` for well-known names controlled by governance
+ - `namesByAddress` - a namespace for each provisioned account
+ - `board` - to map objects to arbitrary names and back
+
+## agoricNames / agoricNamesAdmin - well known names
+
+To look up well-known objects, such as the BLD brand:
```
Comment (dckc) [inline main/guides/integration/name-services.md:10]:
```suggestion

Well-known names are chosen by [BLD staker governance](https://community.agoric.com/t/about-the-governance-category/15).

To look up well-known objects, such as the BLD brand:
```

File: main/guides/integration/name-services.md (hunk 1 above)
Comment (dckc) [inline main/guides/integration/name-services.md:42]:
```suggestion
const { instance } = await E(zoe).startInstance(lemonadeStand);
const { brands: { Lemonade: lemonadeBrand } } = await E(zoe).getTerms(instance);
const brandAdmin = E(agoricNamesAdmin).lookupAdmin('brand');
await E(brandAdmin).update('Lemonade', lemonadeBrand);
```

File: main/guides/integration/name-services.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+const { instance } = await E(zoe).startInstance(myNewContract);
+const { brands: { Thing: thingBrand } } = await E(zoe).getTerms(instance);
+const brandAdmin = E(agoricNamesAdmin).lookupAdmin('brand');
+await E(brandAdmin).update('Thing', thingBrand);
+```
+
+### agoricNames in vstorage
+
+The `NameAdmin` interface has an `onUpdate` method for registering a callback.
+This is used to reflec the brand entries, that is: `E(E(agoricNames).lookup('brand')).entries()`
```
Comment (dckc) [inline main/guides/integration/name-services.md:48]:
```suggestion
This is used to reflect the brand entries, that is: `E(E(agoricNames).lookup('brand')).entries()`
```

File: main/guides/integration/name-services.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+into vstorage at `published.agoricNames.brand` each time they change.
+The entries for `instance` and the other keys of `agoricNames` are likewise reflected under `published.agoricNames`.
+See [marshalling amounts and instance](../getting-started/contract-rpc.md#marshalling-amounts-and-instances)
+for details on the format of the data stored in vstorage.
+
+_TODO: IOU a screenshot of the vstorage viewer_
```
Comment (dckc) [inline main/guides/integration/name-services.md:54]:
```suggestion
```

**Pull request merged**